gnowsys-ndf/gnowsys_ndf/ndf/views/mio.py

```diff
@@ -0,0 +1,432 @@
+		return error_message["already exists"]
+	else:
+		p.fill_gstystem_values(name = name,member_of=[gst_page._id],created_by = created_by,
+			content = content,group_set=[gst_group._id])
+		p.origin.append({'mio':m_id})
+		try:
+			p.save()
+		except:
+			print "in except"
+		return str(p._id);
+
+def send_page(to_user,page_name,page_content,subject,m_id,ref):
+        msg = MIMEMultipart('alternative')
+        msg["Message-ID"] = email.utils.make_msgid()
+        msg["In-Reply-To"] = m_id
+
+        if(ref==None):
+                msg["References"] = m_id
+        else:
+                msg["References"] = ref+"\n"+m_id
+
+        msg['Subject'] = subject
+        msg['From'] = "ps.mio.bits@gmail.com"
+        msg['To'] = to_user
+        
+        context = {'pg_name': page_name,'pg_content': page_content}
+    	Html = render(None, 'ndf/mio_index.html', context)
+       	html = str(Html)
+       	html = html.strip('Content-Type: text/html; charset=utf-8\n')	
+        msg.attach(MIMEText(html, 'html'))
+        smtpObj=smtplib.SMTP('smtp.gmail.com', 587)
+        smtpObj.ehlo()
+        smtpObj.starttls()
+        smtpObj.login('ps.mio.bits@gmail.com','1Guesswhat')
+        smtpObj.sendmail('ps.mio.bits@gmail.com',to_user,msg.as_string())
+        smtpObj.quit()
+
+def update_page(name,content,id):
+	gst_page = node_collection.one({u'_id':ObjectId(id)})
+	gst_page.name = unicode(name)
+	gst_page.altnames = unicode(name)
+	gst_page.content = unicode(content)
+
+	try:
+		gst_page.save()
+	except:
+		print 'in except'
+
+def upload(gp_name,filename,name,author,content):
+	home_grp = node_collection.one({'_type': "Group", 'name':gp_name})
+	file_gst = node_collection.one({'_type': "GSystemType", 'name': u'File'})
+
+	each_img_name = filename
+	each_img_name_wo_ext = each_img_name.split('.')[0]
+	path = './attachment'
+
+	try:
+		file_obj_in_str = open(path+'/'+each_img_name,'r+')
+		#file_obj_in_str = open('./attachment/helloworld.txt','r+')
+	except Exception as e:
+		print "can't open" + str(e)
+
+	img_file = io.BytesIO(file_obj_in_str.read())
+	img_file.seek(0)
+	fh_obj = filehive_collection.collection.Filehive()
+	filehive_obj_exists = fh_obj.check_if_file_exists(img_file)
+	file_gs_obj = None
+
+	if not filehive_obj_exists:
+		file_gs_obj = node_collection.collection.GSystem()
+
+		file_gs_obj.fill_gstystem_values(
+										request=None,
+										name=unicode(name),
+										group_set=[home_grp._id],
+										# language=language,
+										uploaded_file=img_file,
+										created_by=author,
+										member_of=[file_gst._id],
+										origin={'user-icon-list-path':str(path)},
+										unique_gs_per_file=True,
+										content_org=content
+										)
+		print "\n\nGS obj id "+str(file_gs_obj._id)+"\n\n"
+		try:
+			# file_gs_obj.save(groupid=home_grp._id)
+			file_gs_obj.save(groupid=home_grp._id)
+		except Exception as e:
+			print 'exception while saving', e
+	else:
+		print "file exists"
+	return 'done'
```

Comment:
**[Convention]:** Possibly, return `Bool (True/False or 1/0)` if you are not returning any data from function. And use same at function calling end.